src/lib/Libifl/auth.c

```diff
@@ -369,31 +369,41 @@ is_valid_encrypt_method(char *method)
  *	tcp_send_auth_req - encodes and sends PBS_BATCH_Authenticate request
  *
  * @param[in] sock - socket descriptor
+ * @param[in] port - parent port in pbs_iff (only used in resvport auth) else 0
+ * @param[in] user - authenticating user name
+ * @param[in] auth_method - auth method name
+ * @param[in] encrypt_method - encrypt method name
  *
  * @return	int
  * @retval	0 on success
  * @retval	-1 on error
  */
 int
-tcp_send_auth_req(int sock, unsigned int port, char *user)
+tcp_send_auth_req(int sock, unsigned int port, char *user, char *auth_method, char *encrypt_method)
 {
 	struct batch_reply *reply;
 	int rc;
-	int am_len = strlen(pbs_conf.auth_method);
-	int em_len = strlen(pbs_conf.encrypt_method);
+	int am_len;
+	int em_len = encrypt_method ? strlen(encrypt_method) : 0;
```

Comment:
Can we do this check after making sure that auth_method != NULL i.e. after the following block of code.

Reply:
does it matter?

Reply:
I was thinking that if auth_method is NULL then we don't need to calculate em_len where we are using strlen to calculate the length.

Reply:
yeah i got that, but that error path will seldom happen and we do not need to really bother much about a seldom happening error path's performance, right?

Reply:
Yes. If it doesn't happen frequently there is no much performance impact. I agree with you.